internal/provider/data_source_application_test.go

```diff
@@ -16,7 +16,7 @@ func TestAccArgoCDApplicationDataSource(t *testing.T) {
 				ExternalProviders: map[string]resource.ExternalProvider{
 					"argocd": {
 						VersionConstraint: "~> 5.0",
```

Comment:
Shouldn't the `VersionConstraint` be `"~> 6.2.0` (or even "~> 7.0)?

Reply:
I did not want to touch any functionality yet. Unsure why we had `~> 5.0` here. Maybe we should completely remove the version constriant?

Reply:
Created a follow-up issue for this:
- https://github.com/argoproj-labs/terraform-provider-argocd/issues/440